rinja_derive/src/generator/expr.rs

```diff
@@ -398,8 +402,10 @@ impl<'a> Generator<'a, '_> {
         buf: &mut Buffer,
         name: &str,
         args: &[WithSpan<'_, Expr<'a>>],
+        generics: &[WithSpan<'_, TyGenerics<'_>>],
         node: Span<'_>,
     ) -> Result<DisplayWrap, CompileError> {
+        ensure_filter_has_no_generics(ctx, name, generics, node)?;
```

Comment:
I'm not a big fan because this call can (and will be) forgotten when we add new filters. I voluntarily moved this check outside of the function to special-case filters which actually were making use of generics so by default, all the others would error.

Reply:
Actually I don't think that the line get be forgotten. Without the line, the argument `generics` is unused, and rust will print a warning.

For functions like `visit_filter()` I like to use a clear dispatch pattern. The brains should be in the dispatch target functions, the dispatcher should be as stupid and as simple as possible. And because an unused argument would render a warning, I don't see any drawbacks in here.

But if you dislike this implementation very much, then I could live is an implementation with more logic inside the dispatcher, too.

Reply:
It's a valid argument although I think centralizing checks when possible is a better approach. And also, sometimes you can just prepend with `_` to do your things and forget about it later on (happened to me :sweat_smile:).

Reply:
Yeay. :D

But another day, tonight it's video game time. :)

Reply:
Hehe, have fun! :)